core/src/OpenACC/Kokkos_OpenACC_ParallelReduce_Range.hpp

```diff
@@ -97,60 +97,66 @@ class Kokkos::Impl::ParallelReduce<Functor, Kokkos::RangePolicy<Traits...>,
   }
 };
 
-#define KOKKOS_IMPL_OPENACC_PARALLEL_REDUCE_DISPATCH_SCHEDULE(REDUCER,         \
-                                                              OPERATOR)        \
-  namespace Kokkos::Experimental::Impl {                                       \
-  template <class IndexType, class ValueType, class Functor>                   \
-  void OpenACCParallelReduce##REDUCER(Schedule<Static>, int chunk_size,        \
-                                      IndexType begin, IndexType end,          \
-                                      ValueType& val, Functor const& afunctor, \
-                                      int async_arg) {                         \
-    /* FIXME_OPENACC FIXME_NVHPC workaround compiler bug (incorrect scope      \
-       analysis)                                                               \
-       NVC++-S-1067-Cannot determine bounds for array - functor */             \
-    auto const functor(afunctor);                                              \
-    if (chunk_size >= 1) {                                                     \
+#define KOKKOS_IMPL_OPENACC_PARALLEL_REDUCE_DISPATCH_SCHEDULE(REDUCER,    \
+                                                              OPERATOR)   \
+  namespace Kokkos::Experimental::Impl {                                  \
+  template <class IndexType, class ValueType, class Functor>              \
+  void OpenACCParallelReduce##REDUCER(Schedule<Static>, int chunk_size,   \
+                                      IndexType begin, IndexType end,     \
+                                      ValueType& aval,                    \
+                                      Functor const& afunctor,            \
+                                      int async_arg) {                    \
+    /* FIXME_OPENACC FIXME_NVHPC workaround compiler bug (incorrect scope \
+       analysis)                                                          \
+       NVC++-S-1067-Cannot determine bounds for array - functor */        \
+    auto const functor(afunctor);                                         \
```

Comment:
`auto const functor(afunctor);` does not cause an error since the current OpenACC implementation does not use `work_tag` for now.

Reply:
Whether you chose to write `auto const functor(afunctor)` or `auto const functor = afunctor;` does not change the type .

Reply:
The type of the variable, `functor` will be the same (`FunctorAdapter`), but its member data (`FuncterAdapter.m_functor`) will be different. (`functor.m_functor` should be the original functor that the user provided, instead of being `FunctorAdapter` wrapper.)

Reply:
I don't think that is true.  You can try https://godbolt.org/z/ae3qPanhx to convince yourself of it.
Both declaration just go through the copy constructor and yield what we want.

Reply:
You are right; both `auto const functor(afunctor)` and `auto const functor = afunctor` have the same effect, having the same type (`FunctorAdapter` type) and the same member type.